### PR TITLE
feat: pkg/util/k8s add ut

### LINKED ISSUE #1015: :four_leaf_clover: `Proposal`: add more ut for pkg/util/k8s
## What Would You Like to Add? Why Is This Needed?
many plugin need to use pkg/util/k8s module, we should add unit test for this module

### PR DESCRIPTION
**Pull request by steinliber**
Signed-off-by: Meng JiaFeng <jiafeng.meng@merico.dev>

## Pre-Checklist

Note: please complete **_ALL_** items in the following checklist.

- [x] I have read through the [CONTRIBUTING.md](https://github.com/devstream-io/devstream/blob/main/CONTRIBUTING.md) documentation.
- [x] My code has the necessary comments and documentation (if needed).
- [x] I have added relevant tests

## Description
add more unit test for pkg/util/k8s package

## Related Issues
close #1015

## New Behavior (screenshots if needed)
image

### CODE REVIEW

**Review by iyear — APPROVED** [review]
LGTM 👏

File: pkg/util/k8s/configmap_test.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+				}}
+				client.clientset = fake.NewSimpleClientset(testConfigMap...)
+			})
+			It("should udpate configmap", func() {
+				currectConfigMap, err := client.clientset.CoreV1().ConfigMaps(
+					namespace).Get(context.Background(), configmapName, metav1.GetOptions{})
+				Expect(err).Error().ShouldNot(HaveOccurred())
+				Expect(currectConfigMap.Data).Should(Equal(data))
+				Expect(currectConfigMap.ObjectMeta.Labels).Should(Equal(labels))
+			})
```
Comment (aFlyBird0) [inline pkg/util/k8s/configmap_test.go:55]: You didn't call `ApplyConfig` so how did you test this function?

File: pkg/util/k8s/configmap_test.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+						Name:      configmapName,
+						Namespace: namespace,
+						Labels:    labels,
+					},
+					Data: data,
+				}}
+				client.clientset = fake.NewSimpleClientset(testConfigMap...)
+			})
+			It("should udpate configmap", func() {
+				currectConfigMap, err := client.clientset.CoreV1().ConfigMaps(
```
Comment (aFlyBird0) [inline pkg/util/k8s/configmap_test.go:50]: typo

File: pkg/util/k8s/configmap_test.go [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+		When("configmap not exist", func() {
+			BeforeEach(func() {
+				configmapName = "not_exist_config"
+			})
+			It("should return not found error", func() {
+				_, err := client.GetConfigMap(namespace, configmapName)
+				Expect(err).Error().Should(HaveOccurred())
+				Expect(errors.IsNotFound(err)).Should(BeTrue())
+			})
+		})
```
Comment (aFlyBird0) [inline pkg/util/k8s/configmap_test.go:68]: Maybe this way is better. (The code is copied from `Context("ApplyConfigMap method", func() {...`, I just edit the prompt and var name)
```go
BeforeEach(func() {
			testConfigMap = []runtime.Object{&v1.ConfigMap{
				ObjectMeta: metav1.ObjectMeta{
					Name:      configmapName,
					Namespace: namespace,
					Labels:    labels,
				},
				Data: data,
			}}
			client.clientset = fake.NewSimpleClientset(testConfigMap...)
		})
		It("should get correct configmap", func() {
			currentConfigMap, err := client.clientset.CoreV1().ConfigMaps(
				namespace).Get(context.Background(), configmapName, metav1.GetOptions{})
			Expect(err).Error().ShouldNot(HaveOccurred())
			Expect(currentConfigMap.Data).Should(Equal(data))
			Expect(currentConfigMap.ObjectMeta.Labels).Should(Equal(labels))
		})
```

Comment (aFlyBird0) [inline pkg/util/k8s/configmap_test.go:68]: Maybe you could add the `Context` when "configmap is exist". (The code is copied from `Context("ApplyConfigMap method", func() {...`, I just edit the prompt and var name)
```go
BeforeEach(func() {
			testConfigMap = []runtime.Object{&v1.ConfigMap{
				ObjectMeta: metav1.ObjectMeta{
					Name:      configmapName,
					Namespace: namespace,
					Labels:    labels,
				},
				Data: data,
			}}
			client.clientset = fake.NewSimpleClientset(testConfigMap...)
		})
		It("should get correct configmap", func() {
			currentConfigMap, err := client.clientset.CoreV1().ConfigMaps(
				namespace).Get(context.Background(), configmapName, metav1.GetOptions{})
			Expect(err).Error().ShouldNot(HaveOccurred())
			Expect(currentConfigMap.Data).Should(Equal(data))
			Expect(currentConfigMap.ObjectMeta.Labels).Should(Equal(labels))
		})
```

**Review by aFlyBird0 — CHANGES_REQUESTED** [review]

**Comment by aFlyBird0** [comment]
Thanks for creating detailed uts for `k8s` pkg. 👏

**Review by daniel-hutao — APPROVED** [review]

**Pull request merged**